frontend/src/app/results/results.component.ts

```diff
@@ -0,0 +1,47 @@
+import {Component, OnInit} from "@angular/core";
+import {Query} from "../models/query";
+import {ResultsService} from '../services/results.service';
+import {ActivatedRoute, Params} from "@angular/router";
+import { MdSnackBar, MdSlider } from '@angular/material';
+
+@Component({
+  selector: 'app-results',
+  templateUrl: './results.component.html'
+})
+export class ResultsComponent implements OnInit {
+
+  private query: Query;
+  private queryID: string;
+  private imgOpacity: number;
+
+  constructor(private route: ActivatedRoute,
+              private resultsService: ResultsService,
+              private mdSnackBar: MdSnackBar) {
+  }
+
+  ngOnInit() {
+    this.route.params
+      .map((params: Params) => params['id'])
+      .subscribe(id => {
+        this.queryID = id;
+        this.getQueryResult();
+      })
+  }
+
+  /**
+   * Get Query in order to display search result
+   */
+  getQueryResult() {
+    this.resultsService.getQuery(this.queryID).subscribe(
```

Comment:
queryID shouldn't be a controller attribute since it is only used here and have not much sense for the controller.

You could pass the ID as a parameter of getQueryResult() instead.